public/app/core/navigation/GrafanaRoute.tsx

```diff
@@ -16,12 +16,14 @@ export class GrafanaRoute extends React.Component<Props> {
     keybindingSrv.initGlobals();
     reportPageview();
     navigationLogger('GrafanaRoute', false, 'Mounted', this.props.match);
+    window.parent.postMessage('iframeNavigate', window.parent.location.origin);
```

Comment:
That is an interesting thought. I guess I assumed it would go in ni/priv since it is using an NI-defined message "iframeNavigate".

Reply:
It looks like the allow_embedding is only used by the back-end so it might be non-trivial to get that value of that flag in the front-end.

Reply:
There's nothing NI-specific in the message or what we're doing here though, right? It looks generic enough to me. If we were to make Grafana publish the message, other folks embedding Grafana in other products would be able to make use of it too I assume?

Reply:
Correct. It is probably worth at least creating a ni/pub branch and leaving a TODO to figure out the conditional logic like you mentioned. In the meantime we can just merge the change into ni/priv for immediate use.

Reply:
To be clear, I'm not suggesting we block merging your branch into `ni/current` on upstreaming this, but I'd just like to start that up in parallel if we think it makes sense. I know we have short-term goals, and it'd be great to have this ready for iteration demos next week. We certainly won't have it upstreamed by then.

Reply:
I think I'll need to create a separate ni/pub/* branch and duplicate the changes there - I think we'll run into the same merge conflict issues when trying to merge into ni/priv from there since it will be based off of ni/current.

Reply:
I think the idea is to merge from `ni/pub/*` directly into `ni/current`. We don't go through `ni/priv` in that case.

Reply:
Right, but if we need the change in the short term we would need to merge into ni/priv as well. In any case, I'll create a separate ni/pub branch with the TODO so we at least have this for the future. It's probably good that we wait to upstream anyways since our iframe-based integration might be going away next quarter.

Reply:
(but for this particular branch and PR where we're pushing it through `ni/priv` in the short-term, we would go through `ni/priv` of course)

Reply:
Man, github has a bad refresh rate for these comment threads. Oh well. We're on the same page.

Reply:
Created ni/pub branch to track these changes.
https://github.com/ni/grafana/commit/e8ac8c161774da726be5092f4809b97b42485962